core/src/test/java/io/grpc/ClientInterceptorsTest.java

```diff
@@ -351,6 +352,33 @@ protected void checkedStart(ClientCall.Listener<RespT> responseListener, Metadat
     ArgumentCaptor<Status> captor = ArgumentCaptor.forClass(Status.class);
     verify(listener).onClose(captor.capture(), any(Metadata.class));
     assertSame(error, captor.getValue().getCause());
+
+    // Make sure nothing bad happens after the exception.
+    ClientCall<?, ?> noop = ((CheckedForwardingClientCall<?, ?>)interceptedCall).delegate();
```

Comment:
Have a link to the Client Interceptors coverage?  The coveralls page link (
https://coveralls.io/builds/3840457/source?filename=core%2Fsrc%2Fmain%2Fjava%2Fio%2Fgrpc%2FClientInterceptors.java
) is dead.


On Wed, Oct 14, 2015 at 5:13 PM, Eric Anderson <notifications@github.com>
wrote:

> In core/src/test/java/io/grpc/ClientInterceptorsTest.java
> <https://github.com/grpc/grpc-java/pull/1118#discussion_r42070674>:
>
> > @@ -351,6 +352,33 @@ protected void checkedStart(ClientCall.Listener<RespT> responseListener, Metadat
> >      ArgumentCaptor<Status> captor = ArgumentCaptor.forClass(Status.class);
> >      verify(listener).onClose(captor.capture(), any(Metadata.class));
> >      assertSame(error, captor.getValue().getCause());
> > +
> > +    // Make sure nothing bad happens after the exception.
> > +    ClientCall<?, ?> noop = ((CheckedForwardingClientCall<?, ?>)interceptedCall).delegate();
>
> ut https://coveralls.io/builds/3840457 for ClientInterceptors is missing.
> :/
>
> Actually, it's not missing. The search is *case sensitive* (doh!)
>
> I added the other methods to the Noop call to have them in one place
> (hopefully /the/ place to check) in case some test accidentally stops
> touching those lines in the future.
>
> In my mind that place sort of already existed with interceptedCall above.
>
> The use of verifyNoMoreInteractions is kind of moot IMO, the point isn't
> so much that it isn't invoked, but that any interaction with it is safe and
> inert.
>
> The verifyNoMoreInteractions was verifying that the interaction was inert.
> For example, cancel() should never fail, so the implementation could
> accidentally be calling call and the test wouldn't notice without the
> extra verify.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-java/pull/1118/files#r42070674>.
>